src/plugins/auto/src/plugin.cpp

```diff
@@ -852,25 +846,35 @@ std::vector<DeviceInformation> Plugin::filter_device_by_model(const std::vector<
     }
     if (stateful_node_names.empty()) {
         // not stateful model
-        return meta_devices;
+        return filter_device;
     }
 
     // disable CPU_HELP and runtime fallback if model is stateful
     disable_startup_runtime_fallback();
 
     auto is_supported_stateful = [&](const std::string& device_name, const ov::AnyMap& config) {
-        auto device_qm = get_core()->query_model(model, device_name, config);
+        auto query_config = config;
+        // remove core property from config, to be checked further if can do this elegantly
```

Comment:
@ilya-lavrenov @riverlijunjie meta plugin may need to query model for next steps(auto/hetero), but the passed property may contain core level properties, what do you suggest in this case? in general we don't touch these properties in meta plugin, but hw plugin query model will throw exception if been queries with cache_dir, batching properties etc..

Reply:
why cache_dir is passed to query_model? 
If user do it, it's his mistake

Reply:
@ilya-lavrenov  user just pass the cache_dir or batch property to compile model, but inside AUTO or hetero, for example, AUTO will call query model to decide on the final candidate list, hetero will call query model to decide on the graph splitting logic, so:
1) we remove the unsupported properties inside plugin, but in this way we will have to touch these properties again in plugin , which I think is not  reasonable under current design logic
2) query model to remove only core properties
3) before query model, we add extra call inside meta plugin to filter only plugin supported properties, similar to current get_supported_properties API, but do not return core properties?

Reply:
let's try avoid calling query model from compiled model, it's expensive especially for large models @yangwang201911

Reply:
Updated.